### PR TITLE
Quick status icons

### PR DESCRIPTION
**Pull request by Frankkwizera**
added battery icons and weather icons

### CODE REVIEW

File: seshdash/utils/model_tools.py [hunk 1]
```diff
@@ -153,3 +157,63 @@ def get_site_measurements(site):
             site_measurements_items.pop(i)
 
     return site_measurements_items
+
+def get_quick_status(sites):
+    """
+    returns battery state and future forecast
+    """
+    key = "8eefab4d187a39b993ca9c875fef6159"
```
Comment (jlark) [inline seshdash/utils/model_tools.py]: what's this key?

File: seshdash/utils/model_tools.py [hunk 2]
```diff
@@ -153,3 +157,63 @@ def get_site_measurements(site):
...
+    battery_limit.sort()
+
+    #weather rules limits
+    weather_limit = Status_Rule.weather_rules.keys()
+    weather_limit.sort()
+
+    results = []
+
+    #getting latest_points
+    for site in sites:
```
Comment (jlark) [inline seshdash/utils/model_tools.py]: let's get sites only for the user

File: seshdash/utils/model_tools.py [hunk 3]
```diff
@@ -153,3 +157,63 @@ def get_site_measurements(site):
...
+        site_dict = {}
+
+        site_dict['site'] = site
+        #site coordinates
+        lat = site.position.latitude
+        lng = site.position.longitude
+
+        #forecast instance
+        instance = ForecastAPI(key,lat,lng)
```
Comment (jlark) [inline seshdash/utils/model_tools.py]: You can assume the weather data is already in the weather model. You don't need to fetch it here. It's automatically fetched in tasks.py

File: seshdash/views.py [hunk 4]
```diff
@@ -749,6 +758,11 @@ def historical_data(request):
         active_site = sites[0]
         context_dict = {}
 
+        #sites witth weather and battery status
+        user_sites =  _get_user_sites(request)
```
Comment (jlark) [inline seshdash/views.py]: WHy do we have this in the vrm_login_form?

**Comment by jlark** [comment]
Hi @Frankkwizera, can you add a screenshot and also please see my comments inline

File: seshdash/utils/model_tools.py (hunk 2 above)
Comment (Frankkwizera) [inline seshdash/utils/model_tools.py]: yeah its only user sites

File: seshdash/utils/model_tools.py [hunk 5]
```diff
@@ -175,22 +173,27 @@ def get_quick_status(sites):
...
-
-        battery_latest_dict = get_latest_point_site(site,'battery_voltage')
+        #site weather_data
+        site_weather_data = Site_Weather_Data.objects.filter(site= site)
+        cloud_cover = []
+        for weather_data in site_weather_data:
+            cloud_cover.append(weather_data.cloud_cover)
+
+        #pop out the prevoius days
+        cloud_cover.pop(0)
```
Comment (jlark) [inline seshdash/utils/model_tools.py]: Is this a reliable way to get rid of the current days?

File: seshdash/utils/model_tools.py [hunk 6]
```diff
@@ -158,12 +158,10 @@ def get_site_measurements(site):
 
     return site_measurements_items
```
Comment (jlark) [inline seshdash/utils/model_tools.py]: @login_required is a good to have here

File: seshdash/utils/model_tools.py [hunk 7]
```diff
@@ -207,7 +210,7 @@ def get_quick_status(sites):
             site_dict['battery'] = color
 
         #finding weather_data color
-        if average_weather_data < weather_limit[0]:
+        if average_cloud_cover < weather_limit[0]:
```
Comment (jlark) [inline seshdash/utils/model_tools.py]: Where is the weather_limit defined?

File: settings_local_example.ini [hunk 8]
```diff
@@ -10,16 +10,16 @@ HOST: hostname
...
+PASSWORD:
 DB: test_db
 
 
 [api]
 #API Keys:
 ENPHASE_KEY: your_enphase_key_ask_admin
 #TODO This should go in the model not here move this when you can
-FORECAST_KEY: your_forecast_io_key
+FORECAST_KEY: 8eefab4d187a39b993ca9c875fef6159
```
Comment (jlark) [inline settings_local_example.ini]: Please remove this from the sample config.

File: seshdash/utils/model_tools.py (hunk 7 above)
Comment (Frankkwizera) [inline seshdash/utils/model_tools.py]: at the top of the function

File: seshdash/utils/model_tools.py (hunk 5 above)
Comment (Frankkwizera) [inline seshdash/utils/model_tools.py]: i can get rid of  them based on dates.

**Comment by Frankkwizera** [comment]
quick_icons

**Comment by jlark** [comment]
looking good. Is it done?

**Comment by Frankkwizera** [comment]
there is a bug mukuyu is supposed to be red due to its soc

**Comment by jlark** [comment]
ok let me know when it's ready for review

**Comment by Frankkwizera** [comment]
qucik_1
done..

File: seshdash/tests/test_edit_site.py [hunk 9]
```diff
@@ -50,6 +50,14 @@ def setUp(self):
                                                     AC_output=15.0,
                                                     AC_Load_in=0.0,
                                                     AC_Load_out=-0.7)
+        #creating weather data model
```
Comment (jlark) [inline seshdash/tests/test_edit_site.py]: DId you write any tests for this?

**Comment by Frankkwizera** [comment]
modified test_edit_site.py which was failing due to the changes.

**Comment by jlark** [comment]
can you aslso write a test for the status icons. 
- Create some sample data for weather and battery stats. 
- Then run your function for calculating stats. 
- Make sure output is what you expect.

File: seshdash/utils/model_tools.py [hunk 10]
```diff
@@ -183,6 +183,8 @@ def get_quick_status(user_sites):
         site_dict['site'] = site
 
         battery_latest_dict = get_latest_point_site(site,'soc')
+        print "for site:" + str(site.site_name)
```
Comment (jlark) [inline seshdash/utils/model_tools.py]: Pllease remove print statment replace with logger.debug if necassary

**Comment by jlark** [comment]
test still failing.

File: seshdash/tests/test_status_icons.py [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+from django.utils import timezone
+from seshdash.utils.model_tools import get_quick_status
+from django.contrib.auth.models import User
+from seshdash.data.db.influx import Influx
+
+class StatusTestCase(TestCase):
+    @override_settings(DEBUG=True)
+    def setUp(self):
+
+        self.db_name = 'test_db'
```
Comment (jlark) [inline seshdash/tests/test_status_icons.py]: Your code is wrong see below, you're not creating the database,

`
        self._influx_db_name = 'test_db'
        self.i = Influx(database=self._influx_db_name)

        try:
            self.i.create_database(self._influx_db_name)
            #Generate random data  points for 24h
        except:
           self.i.delete_database(self._influx_db_name)
           sleep(1)
           self.i.create_database(self._influx_db_name)
           pass
`

**Pull request merged**